Nonetype Entires Break Individual Image Metadata

`None` values in Metadata can cause validation errors:

```
def mk_rcs(**kwargs):
  ...
  meta.rcs["bank"] = kwargs.get("bank", None)
  meta.rcs["led"] = kwargs.get("led", None)
  ...
```

Multiple files then added into a mosaic via:

`wfi_mosaic_model.append_individual_image_meta(wfi_imageX.meta)`

Then trying to validate:

`wfi_mosaic_model.validate()`

Yields:
```
dtype = dtype('O'), include_byteorder = True, override_byteorder = None

    def numpy_dtype_to_asdf_datatype(dtype, include_byteorder=True, override_byteorder=None):
        dtype = np.dtype(dtype)
        if dtype.names is not None:
            fields = []
            for name in dtype.names:
                field = dtype.fields[name][0]
                d = {}
                d["name"] = name
                field_dtype, byteorder = numpy_dtype_to_asdf_datatype(field, override_byteorder=override_byteorder)
                d["datatype"] = field_dtype
                if include_byteorder:
                    d["byteorder"] = byteorder
                if field.shape:
                    d["shape"] = list(field.shape)
                fields.append(d)
            return fields, numpy_byteorder_to_asdf_byteorder(dtype.byteorder, override=override_byteorder)
    
        if dtype.subdtype is not None:
            return numpy_dtype_to_asdf_datatype(dtype.subdtype[0], override_byteorder=override_byteorder)
    
        if dtype.name in _datatype_names:
            return dtype.name, numpy_byteorder_to_asdf_byteorder(dtype.byteorder, override=override_byteorder)
    
        if dtype.name == "bool":
            return "bool8", numpy_byteorder_to_asdf_byteorder(dtype.byteorder, override=override_byteorder)
    
        if dtype.name.startswith("string") or dtype.name.startswith("bytes"):
            return ["ascii", dtype.itemsize], "big"
    
        if dtype.name.startswith("unicode") or dtype.name.startswith("str"):
            return (
                ["ucs4", int(dtype.itemsize / 4)],
                numpy_byteorder_to_asdf_byteorder(dtype.byteorder, override=override_byteorder),
            )
    
        msg = f"Unknown dtype {dtype}"
>       raise ValueError(msg)
E       ValueError: Unknown dtype object

/opt/anaconda3/envs/romancal/lib/python3.12/site-packages/asdf/tags/core/ndarray.py:148: ValueError
```

--- a/changes/485.removal.rst
+++ b/changes/485.removal.rst
@@ -1,0 +1,1 @@
+Remove MosaicModel.append_individual_meta

--- a/src/roman_datamodels/datamodels/_datamodels.py
+++ b/src/roman_datamodels/datamodels/_datamodels.py
@@ -9,11 +9,9 @@ This module provides all the specific datamodels used by the Roman pipeline.
 import copy
 import logging
 
-import asdf
 import astropy.table.meta
 import numpy as np
 from astropy.modeling import models
-from astropy.table import QTable
 
 from .. import stnode
 from ._core import DataModel
@@ -119,77 +117,6 @@ class _RomanDataModel(_DataModel):
 
 class MosaicModel(_RomanDataModel):
     _node_type = stnode.WfiMosaic
-
-    def append_individual_image_meta(self, meta):
-        """
-        Add the contents of meta to the appropriate keyword in individual_image_meta as an astropy QTable.
-
-        Parameters
-        ----------
-        meta : roman_datamodels.stnode._node.DNode or dict
-            Metadata from a component image of the mosiac.
-        """
-
-        # Convert input to a dictionary, if necessary
-        if not isinstance(meta, dict | asdf.lazy_nodes.AsdfDictNode):
-            meta_dict = meta.to_flat_dict()
-        else:
-            meta_dict = meta
-
-        # Storage for keys and values in the base meta layer
-        basic_cols = []
-        basic_vals = []
-
-        # Sift through meta items to place in tables
-        for key, value in meta_dict.items():
-            # Skip wcs objects
-            if key == "wcs":
-                continue
-
-            # Keys that are themselves Dnodes (subdirectories)
-            # neccessitate a new table
-            if isinstance(value, dict | asdf.tags.core.ndarray.NDArrayType | QTable | asdf.lazy_nodes.AsdfDictNode):
-                continue
-
-            if isinstance(value, stnode.DNode):
-                # Storage for keys and values
-                subtable_cols = []
-                subtable_vals = []
-
-                # Loop over items within the node
-                for subkey, subvalue in meta_dict[key].items():
-                    # Skip ndarrays and QTables
-                    if isinstance(subvalue, asdf.tags.core.ndarray.NDArrayType | QTable):
-                        continue
-
-                    subtable_cols.append(subkey)
-                    subtable_vals.append(
-                        [str(subvalue)]
-                        if isinstance(subvalue, list | dict | asdf.lazy_nodes.AsdfDictNode | asdf.lazy_nodes.AsdfListNode)
-                        else [subvalue]
-                    )
-
-                # Skip this Table if it would be empty
-                if subtable_vals:
-                    # Make new Keyword Table if needed
-                    if (key not in self.meta.individual_image_meta) or (
-                        self.meta.individual_image_meta[key].colnames == ["dummy"]
-                    ):
-                        self.meta.individual_image_meta[key] = QTable(names=subtable_cols, data=subtable_vals)
-                    else:
-                        # Append to existing table
-                        self.meta.individual_image_meta[key].add_row(subtable_vals)
-            else:
-                # Store Basic keyword
-                basic_cols.append(key)
-                basic_vals.append([str(value)] if isinstance(value, list | asdf.lazy_nodes.AsdfListNode) else [value])
-
-        # Make Basic Table if needed
-        if self.meta.individual_image_meta.basic.colnames == ["dummy"]:
-            self.meta.individual_image_meta.basic = QTable(names=basic_cols, data=basic_vals)
-        else:
-            # Append to existing basic table
-            self.meta.individual_image_meta.basic.add_row(basic_vals)
 
 
 class ImageModel(_RomanDataModel):

--- a/tests/test_models.py
+++ b/tests/test_models.py
@@ -7,7 +7,7 @@ import pytest
 from asdf.exceptions import ValidationError
 from astropy import units as u
 from astropy.modeling import Model
-from astropy.table import QTable, Table
+from astropy.table import Table
 from numpy.testing import assert_array_equal
 
 from roman_datamodels import datamodels, stnode, validate
@@ -734,47 +734,6 @@ def test_make_level3_mosaic():
 
     # Test validation
     wfi_mosaic_model = datamodels.MosaicModel(wfi_mosaic)
-    assert wfi_mosaic_model.validate() is None
-
-
-# WFI Level 3 Mosaic tests
-def test_append_individual_image_meta_level3_mosaic():
-    wfi_mosaic = utils.mk_level3_mosaic(shape=(8, 8))
-    wfi_mosaic_model = datamodels.MosaicModel(wfi_mosaic)
-
-    wfi_image1 = utils.mk_level2_image(shape=(8, 8))
-    wfi_image2 = utils.mk_level2_image(shape=(8, 8))
-    wfi_image3 = utils.mk_level2_image(shape=(8, 8))
-
-    wfi_image1.meta.program.investigator_name = "Nancy"
-    wfi_image2.meta.program.investigator_name = "Grace"
-    wfi_image3.meta.program.investigator_name = "Roman"
-    wfi_image3_model = datamodels.ImageModel(wfi_image3)
-
-    wfi_mosaic_model.append_individual_image_meta(wfi_image1.meta)
-    wfi_mosaic_model.append_individual_image_meta(wfi_image2.meta.to_flat_dict())
-    wfi_mosaic_model.append_individual_image_meta(wfi_image3_model.meta)
-
-    # Test that basic is a QTable
-    assert isinstance(wfi_mosaic_model.meta.individual_image_meta.basic, QTable)
-
-    # Test that each image entry in the instrument table contains the same instrument name
-    assert wfi_mosaic_model.meta.individual_image_meta.instrument["name"][0] == "WFI"
-    assert (
-        wfi_mosaic_model.meta.individual_image_meta.instrument["name"][0]
-        == wfi_mosaic_model.meta.individual_image_meta.instrument["name"][1]
-    )
-    assert (
-        wfi_mosaic_model.meta.individual_image_meta.instrument["name"][1]
-        == wfi_mosaic_model.meta.individual_image_meta.instrument["name"][2]
-    )
-
-    # Test that each image entry in the program table contains the correct (different) PI name
-    assert wfi_mosaic_model.meta.individual_image_meta.program["investigator_name"][0] == "Nancy"
-    assert wfi_mosaic_model.meta.individual_image_meta.program["investigator_name"][1] == "Grace"
-    assert wfi_mosaic_model.meta.individual_image_meta.program["investigator_name"][2] == "Roman"
-
-    # Test that the mosaic validates with IIM filled
     assert wfi_mosaic_model.validate() is None
 
 

